### DISCUSSION TITLE
New template for svelte

### DISCUSSION THREAD

**Discussion by donpedro** [TotallyInformation/node-red-contrib-uibuilder#183] [Show and tell]
Thanks for your work on this project, I'm really enjoying it!

I've just created a new template called [svelte-uibuilder](https://github.com/donpedro/svelte-uibuilder), based on [svelte-basic](https://github.com/TotallyInformation/node-red-contrib-uibuilder/tree/v6.0.0/templates/svelte-basic); . It is suitable for loading directly into uibuilder with degit.

The goal was to use the latest version of svelte and all of the current best practices and tooling. Everything looks great, except unfortunately I couldn't get Vite's dev mode to play nice with the Node-RED/uibuilder server; it won't auto-reload the browser in that scenario. It does automatically rebuild when changes are made, you just have to then reload the browser.

**Comment by TotallyInformation** [comment]
Cool, thanks for this and sorry about the delayed response. I have added this to the [External Templates Catalogue](https://github.com/TotallyInformation/node-red-contrib-uibuilder/wiki/External-Templates-Catalogue) in the WIKI.

In regard to the dev server, I was sure that was working for me. Just leave the Svelte dev server running but load the page from your usual node-red URL.

As a work-around, you can also create a small node-red flow that will watch for changes and sends your uibuilder node a simple reload message when it detects changes. See https://totallyinformation.github.io/node-red-contrib-uibuilder/#/browser-refresh for details.
